src/ng/services/table.js

```diff
@@ -1,22 +1,143 @@
+class Dom {
+	column() {
+		return {cells: []};
+	}
+
+	row() {
+		return {cells: []};
+	}
+
+	cell() {
+		return null;
+	}
+
+	get view() {
+		return null;
+	}
+}
+class TableDom extends Dom {
+	constructor(element) {
+		super();
+		this.element = element;
+	}
+
+	static get empty() {
+		return new Dom();
+	}
+
+	get view() {
+		return this.element;
+	}
+
+	column(index) {
+		return new ColumnDom(this.element, index);
+	}
+
+	row(index) {
+		const rows = this.element.rows;
+		if (index >= 0 && index < rows.length) {
+			return new RowDom(rows[index]);
+		}
+		return null;
+	}
+
+	cell(row, column) {
+		const rows = this.element.rows;
+		row = row < 0 ? 0 : row;
```

Comment:
Also, Kate, can you add tests for that file, please? Cause that will be essentual part of the grid